src/ledger/orderbook.ts

```diff
@@ -50,7 +51,11 @@ function formatBidsAndAsks(
   // for asks: lowest quality => lowest totalPrice/quantity => lowest price
   // for both bids and asks, lowest quality is closest to mid-market
   // we sort the orders so that earlier orders are closer to mid-market
-  const orders = _.sortBy(offers, 'quality').map(parseOrderbookOrder)
+
+  const orders = offers.sort((a, b) => {
+    return (new BigNumber(a.quality)).minus(b.quality).toNumber()
```

Comment:
So this was just sorting quality as a string before ?

Reply:
If the actual ripple API already returns the order book in the correct order is this sorting actually necessary? Referring to a discussion in #766

Reply:
It was sorting as a string.

I am not sure whether sorting is necessary. Reading the code, I see that ripple-lib makes multiple requests in order to fetch the reverse offers (where base/counter are swapped) and then combined the offers together. When there are paginated results, it fetches all the pages (not great) and returns all the data together.

Reply:
+1, I'd expect the lib to return them in the order they were given by the API, by default. Especially if these responses are paged.

Reply:
I don't think any sorting is required. Rippled already returns them sorted by quality. In case of paged data (when "limit" is omitted or is a large number) they should only be connected in sequence.

Reply:
What do you mean by “connected in sequence”?

Reply:
Rippled API for orderbook returns an array of Offer Objects (https://developers.ripple.com/offer.html). The array is already sorted with increasing/decreasing quality. If paging, you'll get the first array of offers and the marker. You'll have to make another book_offers request to rippled where you'll get the 2nd page of offers objects in an array and so on until you receive no marker in the answer. Just concatenate (Array.concat) the arrays in subsequent pages to obtain the full orderbook.

Reply:
@intelliot If you want the full book (bids and asks) yes, because rippled "book_offers" only returns one leg.